katdal/chunkstore_s3.py

```diff
@@ -156,24 +140,68 @@ def __call__(self, r):
         return r
 
 
-class _Multipart(object):
-    """Allow a sequence of bytes-like objects to be used as a request body.
+def _auth_factory(url, token=None, credentials=None):
+    """Turn either JWT token or AWS credentials into a requests auth handler."""
+    if token is not None and credentials is not None:
+        raise ValueError('Cannot specify both token and credentials')
+    if token is not None:
+        parsed = urllib.parse.urlparse(url)
+        # The exception for 127.0.0.1 lets the unit test work
+        if parsed.scheme != 'https' and parsed.hostname != '127.0.0.1':
+            raise StoreUnavailable('Auth token may only be used with https')
+        return _BearerAuth(token)
+    elif credentials is not None:
+        if not botocore:
+            raise StoreUnavailable('Passing credentials requires botocore to be installed')
+        return _AWSAuth(credentials)
```

Comment:
Add an explicit `return None` when neither type of auth is provided: PEP 8 says

> Be consistent in return statements. Either all return statements in a function should return an expression, or none of them should. If any return statement returns an expression, any return statements where no value is returned should explicitly state this as return None, and an explicit return statement should be present at the end of the function (if reachable).

Reply:
Fixed.